spec/system/admin/products_v3/products_spec.rb

```diff
@@ -165,18 +165,10 @@ def expect_other_columns_visible
       end
 
       it "displays an on hand count in a span for each product" do
-        within(:xpath, '//*[@id="products-form"]/table/tbody[1]/tr[1]/td[5]/div') do
-          expect(page).to have_content "On demand"
-        end
-        within(:xpath, '//*[@id="products-form"]/table/tbody[2]/tr[1]/td[5]/div') do
-          expect(page).to have_content "20" # displays the total stock
-        end
-        within(:xpath, '//*[@id="products-form"]/table/tbody[2]/tr[2]/td[5]/div') do
-          expect(page).to have_content "16" # displays the stock for variant_2
-        end
-        within(:xpath, '//*[@id="products-form"]/table/tbody[2]/tr[3]/td[5]/div') do
-          expect(page).to have_content "4"  # displays the stock for variant_3
-        end
+        expect(page).to have_content "On demand"
+        expect(page).to_not have_content "20" # does not display the total stock
+        expect(page).to have_content "16" # displays the stock for variant_2
+        expect(page).to have_content "4"  # displays the stock for variant_3
```

Comment:
I think that the previous selectors were to avoid false passes of the spec. Imagine this spec running with _Enterprise 4_. It would show the number 4 even without displaying that stock.

Reply:
Do you think we should revert to the xpath code? Maybe an alternative would be to include the count: attribute, or make the numbers a bit more specific, like 111, 222, 333 - I know, in theory, we could still have Enterprise 111, but maybe more unlikely.

What do you think @mkllnk @dacook ?

Reply:
Yes, I would go back to the xpath. If this was a more common use case then I would create helpers for that. Aren't there some helpers for accessing table rows by a displayed name, for example? That would be better.

Reply:
Yes, in fact the inputs have labels associated with them, so something this should work:
```
within row_containing_name("variant name") do
  expect(page).to have_field "On demand", checked: true
  expect(page).to_not have_field "On Hand", with: 20
end
```
(hmm, we need to fix that inconsistent capitalisation..)

Reply:
Sorry, that was targeting the fields inside the popout.

If we want to check the display, we can target the button (which opens the popout):
```
within row_containing_name("variant name") do
  expect(page).to have_button "On Hand", with: "On demand"
  expect(page).to_not have_button "On Hand", with: 20
end
```

Reply:
Even better might be to say something like this. But as far as I know Capybara doesn't have a convenient way to check all text (including button labels). We could make a helper for that, but it's not worth the bother..

```
within row_containing_name("variant name") do
  within column("On Hand") do
    expect(page).to have_text "On demand"
    expect(page).to_not have_text "20"
  end
end
```